packages/expo-notifications/android/src/test/java/expo/modules/notifications/service/delegates/ExpoHandlingDelegateTest.kt

```diff
@@ -0,0 +1,108 @@
+package expo.modules.notifications.service.delegates
+
+import android.os.Parcel
+import androidx.test.core.app.ApplicationProvider
+import expo.modules.notifications.notifications.NotificationManager
+import expo.modules.notifications.notifications.interfaces.NotificationTrigger
+import expo.modules.notifications.notifications.model.Notification
+import expo.modules.notifications.notifications.model.NotificationContent
+import expo.modules.notifications.notifications.model.NotificationRequest
+import expo.modules.notifications.service.NotificationsService
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.spyk
+import io.mockk.verify
+import org.json.JSONObject
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class ExpoHandlingDelegateTest {
```

Comment:
I would prefer to use fakes over mocking here, but I found this rather challenging without refactoring out static method calls and non-injected dependencies from `ExpoHandlingDelegate`. We also have to use `Robolectric` here given the `Context` dependency.

The use of spies and mocks here allows me to limit the changes required

Reply:
Using spies and mocks is ok. We use them extensively in other packages' tests. `RobolectricTestRunner` is also a good choice 👍